### DISCUSSION TITLE
Best practice to load a trace with multiple files

### DISCUSSION THREAD

**Discussion by guykhazma** [ben-manes/caffeine#807] [Q&A] [answered]
Hi,

What is the best practice to load a trace which has many files?
The config file enables to specify multiple files but it seems that you have to list them explicitly.
I am currently using a [simple workaround](https://github.com/ben-manes/caffeine/compare/master...guykhazma:caffeine:traceread?expand=1) which lists paths which are folders.

Thanks!

**Comment by ben-manes** [comment, marked as answer]
A directory might be listed in a variety of orders so I wasn’t sure if that made sense to support. Instead I would list, copy into an editor, and multi-edit into usable format. Then it’s easy to run, especially if you edit the conf file.

To handle [archived files](https://github.com/guykhazma/caffeine/blob/c963b659fdebf64cf9b2682ed932185a86436abe/simulator/src/main/java/com/github/benmanes/caffeine/cache/simulator/parser/AbstractTraceReader.java#L118), e.g. tar.gz with a single entry, did mean multiple had to be handled. That concats in iteration order.

The simulator is meant to be hacked on so changing it’s code to suit your needs is a good option. Anything useful you are welcome to try and upstream, but there’s no expectation as just an analysis tool.

**Reply by guykhazma** [reply]
I see, thank you very much for the quick response!
